### ISSUE TITLE
Dirty string with ampersands

### ISSUE THREAD

**Issue by Qrzysio** [cocur/slugify#226]
How about a bit more dirty string with ampersands after `htmlspecialchars()`?

`$str = '*** ____ &quot;Hello world&quot; Anything ___ *** - something';`

The above gives us `quot-hello-world-quot-anything-something`. Shouldn't the `slugify` cut everything between `&` and `;`?

**Comment by florianeckerstorfer** [comment]
No, it should not. The purpose of this library is to replace special characters with `-`, not clean up strings. Why are you running htmlspecialchars() before passing the string to `slugify`?
